biz.aQute.bndlib/src/aQute/bnd/cdi/CDIAnnotations.java

```diff
@@ -90,8 +89,7 @@ public boolean analyzeJar(Analyzer analyzer) throws Exception {
 				Discover discover = findDiscoveryMode(beansResource);
 
 				jar.getResources()
```

Comment:
You need to take care to close certain Jar objects which may be created above by Jar.fromResource. You may want to close them here instead of delegating to the Processor with addClose. See

https://github.com/bndtools/bnd/blob/a3e9bdb338c8f7c406f7e98219d83f3f268b92df/biz.aQute.bndlib/src/aQute/bnd/osgi/Analyzer.java#L2540-L2544